### PR TITLE
Feature/edit profile modal

### PR DESCRIPTION
**Pull request by OrdainedRat**
Finally done after having rebasing issues :)

## Description

Utilizing the backend branch feature/update_profile (link to his PR below) made by @WaylayOvercast  for this specific task, we added function to the edit profile modal that already exists in the application. It allows users to update several points of their profile (like their name, email, etc.) and the change happens in real time in the component without a refresh. We anticipate some bugs to come from this as a product of the changing state requiring a refresh in other components. 

https://github.com/BloomTech-Labs/underdog-devs-be-a/pull/95

#### Video Link

https://www.loom.com/share/68d70b0de91240a9b3e287953f4769a5)

#### Video Part 2

https://www.loom.com/share/cd84e68aaa0447b18c6119b6462ab3a0

#### Trello Link

<blockquote class="trello-card"><a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;REKnyFuH&#x2F;204-as-a-user-i-want-to-be-able-to-update-my-account-information-in-profile-settings">As a user, I want to be able to update my account information in profile settings</a></blockquote><script src="https://p.trellocdn.com/embed.min.js"></script>

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by robstetson — APPROVED** [review]
Was great to work together to get this functionality working and rendering properly! All around great work!

**Review by jonathan-mcgraw — APPROVED** [review]
Brilliant, sir. It reflects the hard work your group was putting into it yesterday.

**Review by justinpeisker — APPROVED** [review]
This is a great step forward in making the profile page functional and editable!  Great work

**Review by arsalan2243 — APPROVED** [review]
this addition of function would be very useful.
good job.

**Review by areccus — APPROVED** [review]
You did a great job explaining in the loom video and that's amazing that the profile component is functional.

File: src/components/pages/Profile/EditProfile.js [hunk 1]
```diff
@@ -1,33 +1,40 @@
...
     setModalOpen(false);
-    console.log(formValues);
-    clearForm(e);
+    axiosWithAuth()
+      .put('/profile', values)
+      .then(res => {
+        form.setFieldsValue(values);
+      })
+      .catch(err => {
+        console.log(err);
```
Comment (ryan-hamblin) [inline src/components/pages/Profile/EditProfile.js:24]: Please remove any `console.log`'s from the code base.

File: src/components/pages/Profile/EditProfile.js [hunk 2]
```diff
@@ -170,38 +164,16 @@ function EditProfile({ userInfo }) {
...
-            </Radio.Group>
-          </Form.Item>
         </Form>
       </Modal>
     </>
   );
 }
 
 const mapStateToProps = state => {
+  console.log(state);
```
Comment (ryan-hamblin) [inline src/components/pages/Profile/EditProfile.js:174]: please remove this `console.log`

File: package.json [hunk 3]
```diff
@@ -11,9 +11,14 @@
     "@craco/craco": "^6.0.0",
     "@okta/okta-react": "^3.0.2",
     "@okta/okta-signin-widget": "^4.1.2",
+    "@surma/rollup-plugin-off-main-thread": "^2.2.3",
```
Comment (ryan-hamblin) [inline package.json:14]: Can you please let me know what all these packages you've installed are being used for. That seems like a LOT of installations and this PR doesn't seem to be using any of them.

**Review by ryan-hamblin — CHANGES_REQUESTED** [review]
I see there was a lot of additions to the `package.json` `dependency` tree. Please let me know what those packages are being used for because I don't see a need for them for this PR. 

Also there are some instances where you need to handle errors properly instead of `console.log`ing them out. 

Please fix and let me know so I can look at this one again! Great work overall!

File: package.json (hunk 3 above)
Comment (OrdainedRat) [reply package.json:14]: I think these came up when I was having issues getting my local running and experimented with it a bit. I'll work on these changes now!

File: src/components/pages/Profile/EditProfile.js (hunk 2 above)
Comment (OrdainedRat) [reply src/components/pages/Profile/EditProfile.js]: Done!

File: src/components/pages/Profile/EditProfile.js (hunk 1 above)
Comment (OrdainedRat) [reply src/components/pages/Profile/EditProfile.js]: I added some functionality to the error handling. I have posted another video in the PR that walks through the couple changes I made to the modal.

**Pull request merged**